R/dataset.R

```diff
@@ -88,6 +88,56 @@ setMethod("notes<-", "CrunchDataset", function (x, value) {
     invisible(setEntitySlot(x, "notes", value))
 })
 
+#' Get and set the market size for Crunch datasets
+#'
+#' Crunch Datasets allow you to set a target population size in order to extrapolate
+#' population estimates from survey percentages. This function lets you work with
+#' the population size and magnitude.
+#' @param x a Crunch Dataset
+#' @param size the target population size
+#' @param magnitude the order of magnitude with which to display the population
+#' size. Must be either `3`, `6`, or `9` for thousands, millions, and billions respectively.
+#' @return `popSize` and `popMagnitude` return the population size or
+#' magnitude. `setPopultion` returns the modified dataset.
+#' @name population
+#' @aliases popSize popMagnitude setPopulation
+NULL
+
+#' @rdname population
+#' @export
+setMethod("popSize", "CrunchDataset", function (x) {
+    return(settings(x)$population$size)
+})
+
+#' @rdname population
+#' @export
+setMethod("popMagnitude", "CrunchDataset", function (x) {
+    return(settings(x)$population$magnitude)
+})
+
+#' @rdname population
+#' @export
+setMethod("setPopulation", "CrunchDataset", function (x, size, magnitude) {
+    pop <- settings(x)$population
+    if (missing(magnitude)) {
+        if (is.null(pop$magnitude)) {
+            stop("Dataset does not have a magnitude, please supply one")
+        }
+        magnitude <- pop$magnitude
+    }
+    if (missing(size)) {
+        if (is.null(pop$size)) {
+            stop("Dataset does not have a population, please supply one")
+        }
+        size <- pop$size
+    }
+    if (!(magnitude %in% c(3, 6, 9))){
+        stop("Magnitude must be either 3, 6, or 9")
+    }
+    settings(x)$population <- list(magnitude = magnitude, size = size)
+    invisible(return(x))
+})
```

Comment:
Is there a way for me to remove a population? Would `setPopulation(ds, NULL, NULL)` do it? When you remove population in whaam, the whole entity `population` is set to NULL (rather than `population: {magnitude: NULL, size: NULL}`, which I suspect would happen with the command here), which we should copy if possible.

Reply:
I updated this to accept and return NULL values.